### PR TITLE
E2E Tests: Adding a test to check CPT templates initialization.

### PR DESCRIPTION
**Pull request by youknowriad**
blocked by #5906

This PR adds test plugins to Gutenberg, these plugins can be used in E2E tests. The test plugins folders includes a single plugin for now to test CPT templates. 

This pattern could be used to fix this #5909

### CODE REVIEW

File: test/e2e/test-plugins/templates.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?php
```
Comment (danielbachhuber) [inline test/e2e/test-plugins/templates.php]: If we're going to have a dozen or more test plugins eventually, does it make sense to have a more verbose file naming scheme than `templates.php`?

Comment (youknowriad) [reply test/e2e/test-plugins/templates.php]: I see us having probably a dozen of plugins here. I'm open to suggestions on the naming. It's not that critical at the moment.

Comment (gziolo) [reply test/e2e/test-plugins/templates.php]: Yes, let's refactor when this becomes an issue. It's going to be easier to come up with a pattern when we have more files there.

**Review by aduth — APPROVED** [review]
👍

**Pull request merged**